library/src/androidTest/java/com/mux/player/CacheDatastoreInstrumentationTests.kt

```diff
@@ -142,39 +148,211 @@ class CacheDatastoreInstrumentationTests {
+        relativePath = "cacheFile",
         lookupKey = "lookupKey",
         downloadedAtUtcSecs = 1L,
         cacheMaxAge = 2L,
         resourceAge = 3L,
-        cacheControl = "cacheControl"
+        cacheControl = "cacheControl",
+        lastAccessUtcSecs = 4L,
+        sizeOnDisk = 1L
       )
       val secondRecord = FileRecord(
         url = "url2",
         etag = "etag2",
-        file = File("cacheFile"),
+        relativePath = "cacheFile",
         lookupKey = "lookupKey",
         downloadedAtUtcSecs = 1L,
         cacheMaxAge = 2L,
         resourceAge = 3L,
-        cacheControl = "cacheControl"
+        cacheControl = "cacheControl",
+        lastAccessUtcSecs = 4L,
+        sizeOnDisk = 1L
       )
 
-      val writeResult1 = datastore.writeRecord(originalRecord)
+      val writeResult1 = datastore.writeFileRecord(originalRecord)
       Assert.assertTrue(
         "First write of record with key ${originalRecord.lookupKey} should succeed",
         writeResult1.isSuccess
       )
-      val writeResult2 = datastore.writeRecord(secondRecord)
+      val writeResult2 = datastore.writeFileRecord(secondRecord)
       Assert.assertTrue(
         "next write of record with key ${secondRecord.lookupKey} should succeed",
         writeResult2.isSuccess
       )
 
-      // todo - read-out the record and ensure it is equal to the second record
+      val readRecord = datastore.readRecordByLookupKey("lookupKey")
+      Assert.assertEquals(
+        "The record read-out should match the last record written with that lookup key",
+        secondRecord, readRecord
+      )
     }
   }
 
+  @Test
+  fun testReadRecord() {
+    fun testTheCase(url: String) {
+      CacheDatastore(appContext).use { datastore ->
+        val originalRecord = FileRecord(
+          url = url,
+          etag = "etag1",
+          relativePath = "cacheFile",
+          lookupKey = datastore.safeCacheKey(URL(url)),
+          downloadedAtUtcSecs = 1L,
+          cacheMaxAge = 2L,
+          resourceAge = 3L,
+          cacheControl = "cacheControl",
+          lastAccessUtcSecs = 4L,
+          sizeOnDisk = 1L
+        )
+        val result = datastore.writeFileRecord(originalRecord)
+        result.getOrThrow() // not part of test, writing is covered elsewhere
+
+        val readRecord = datastore.readRecordByUrl(url)
+        Assert.assertEquals(
+          "The record should be the same after writing and reading",
+          originalRecord, readRecord
+        )
+      }
+    }
+
+    testTheCase("https://www.mux.com/any/path")
+    testTheCase("https://chunk-gcp-us-east4-vop1.cfcdn.mux.com/v1/chunk/vSIm02ye02gC7NasaSqE5zGP4lN3UJ01Iw01gOd01PapjUbeWza9NSOI02cpcAa02f5Kfh78vqhiWwVCk01bpcumXT4jQbfDJGBzQ02ygzY02QIMiTQuw/10.ts?skid=default&signature=NjVjZDQ1ZjBfNDYyNWYwODAxZmIzZTQ4YzU2YmQyYTZmZDhkNmYyYWQ2YjkxYmVkZmJkNThkOTBkOWRkYmU3NmRhNDVhYWY5OQ==&zone=0&vsid=z3MOq02sdo99wTURNGGxQJgKEk4qHbLSY4C8HfvZTbRPNGT0029u56MOSv8xlmJSior66tll9YK98")
```

Comment:
issue: I believe this will expire due to signature and/or vsid query params.

Reply:
The resource is never requested, but I'll change the base to make that more clear